packages/node/test/unit/dev.test.ts

```diff
@@ -29,27 +29,87 @@ function testForkDevServer(
     workPath: resolve(__dirname, '../dev-fixtures'),
     entrypoint,
     devServerPath: resolve(__dirname, '../../dist/dev-server.mjs'),
-    printLogs: options.printLogs,
   });
 }
 
-(NODE_MAJOR < 18 ? test.skip : test)(
-  'web handlers for node runtime',
-  async () => {
-    const child = testForkDevServer('./web-handlers-node.js');
-    try {
-      const result = await readMessage(child);
-      if (result.state !== 'message') {
-        throw new Error('Exited. error: ' + JSON.stringify(result.value));
-      }
+const teardown: any = [];
+afterAll(() => Promise.all(teardown.map((fn: any) => fn())));
+
+async function withHttpServer(hander: (req: any, res: any) => void) {
+  const server = createServer(hander);
+  teardown.push(promisify(server.close.bind(server)));
+  const address = await listen(server, { port: 0, host: '127.0.0.1' });
+  return address.toString();
+}
+
+async function withDevServer(
+  entrypoint: string,
+  fn: (url: string) => Promise<void>,
+  { runningTimeout }: { runningTimeout?: number } = {}
+) {
+  const child = testForkDevServer(entrypoint);
+
+  const result = await readMessage(child);
+  if (result.state !== 'message') {
+    throw new Error('Exited. error: ' + JSON.stringify(result.value));
+  }
+  const { address, port } = result.value;
+  const url = `http://${address}:${port}`;
 
-      const { address, port } = result.value;
+  const start = Date.now();
 
-      {
-        const response = await fetch(
-          `http://${address}:${port}/api/web-handlers-node`,
-          { method: 'GET' }
-        );
+  return fn(url).finally(async () => {
+    const elapsed = Date.now() - start;
+    if (runningTimeout) await setTimeout(runningTimeout - elapsed);
+    child.send('shutdown', err => {
+      if (err) child.kill(9);
+    });
+    await once(child, 'exit');
+  });
+}
+
+(NODE_MAJOR < 18 ? describe.skip : describe)('web handlers', () => {
```

Comment:
@TooTallNate right now CI is only running for Node.js v16, so ALL this test file is skipped; How we can setup a Github Action job just for ensure to run this file using Node.js v18?